### DISCUSSION TITLE
Auto-Restart on device boot-up

### DISCUSSION THREAD

**Discussion by saumya-banthia** [DesktopECHO/Pi-hole-for-Android#37] [Ideas]
Thanks for building this to-go version of pi hole. I was wondering if we can build a functionality for the app to auto-restart (since this app can use root privileges), in case it is deployed on something that is not battery or UPS backed (like an Android TV box in my case).

(P.S. Many cheaper Android TV boxes come pre-rooted)

**Comment by DesktopECHO** [comment]
Hi, 

- Click the Settings menu ≡ at the top left
- Click Settings
- Scroll down and place a checkmark next to "Autostart"

Pi-hole will now start-up automaticallly with your device.

**Reply by saumya-banthia** [reply]
Thanks for this, I tried these steps. The only thing I changed is the auto start delay to about 5 seconds but when the app starts, it just displays this msg in the log and nothing after that (waited about 5-10 more mins before retrying with the delay set to 30 seconds, but the same thing happens.):

`>>> start`
`The container is not mounted.`
`<<< start`

Is there a way to force the container to mount upon app startup?

**Reply by DesktopECHO** [reply]
Sorry missed this message from way back.
Normally it does mount and boot with that setting enabled.
Perhaps try uninstalling and reinstall the app and/or revoke root privileges and let the app re-ask for root permission.
